File: app/src/main/java/io/neoterm/view/eks/TextButton.java
package io.neoterm.view.eks;

import android.view.View;

import io.neoterm.view.ExtraKeysView;

/**
 * @author kiva
 */

public class TextButton extends ExtraButton {
    private boolean withEnter = false;

    public TextButton(String text) {
        this(text, false);
    }

    public TextButton(String text, boolean withEnter) {
        this.buttonText = text;
        this.withEnter = withEnter;
    }

    @Override
    public void onClick(View view) {
        sendKey(view, buttonText);
        if (withEnter) {
            sendKey(view, "\n");
        }
    }
}

File: app/src/main/java/io/neoterm/view/eks/TextButton.kt
package io.neoterm.view.eks

import android.view.View

/**
 * @author kiva
 */

open class TextButton @JvmOverloads constructor(text: String, withEnter: Boolean = false) : ExtraButton() {
    private var withEnter = false

    init {
        this.buttonText = text
        this.withEnter = withEnter
    }

    override fun onClick(view: View) {
        ExtraButton.sendKey(view, buttonText!!)
        if (withEnter) {
            ExtraButton.sendKey(view, "\n")
        }
    }
}
